titus-ext/aws/src/main/java/com/netflix/titus/ext/aws/AwsInstanceCloudConnector.java

```diff
@@ -223,6 +227,37 @@ public ResourceDimension getInstanceTypeResourceDimension(String instanceType) {
                 });
     }
 
+    @Override
+    public Mono<Instance> getInstance(String instanceId) {
+        DescribeInstancesRequest req = new DescribeInstancesRequest()
+                .withInstanceIds(instanceId)
+                .withSdkRequestTimeout((int) configuration.getAwsRequestTimeoutMs());
+
+        Mono<DescribeInstancesResult> instancesResult = AwsReactorExt.toMono(req, ec2Client::describeInstancesAsync);
+        return instancesResult
+                .filter(isr -> isr.getReservations() != null &&
+                        !isr.getReservations().isEmpty() &&
+                        !isr.getReservations().get(0).getInstances().isEmpty())
+                .map(isr -> {
+                    Reservation reservation = isr.getReservations().get(0);
+                    return toInstance(reservation.getInstances().get(0));
+                })
+                .onErrorResume(e -> {
+                    if (e instanceof AmazonEC2Exception) {
+                        AmazonEC2Exception ec2Exception = (AmazonEC2Exception) e;
+                        logger.warn("Exception fetching AWS instance {}", instanceId, ec2Exception);
```

Comment:
Move logger after 400 check, as the latter is ok case? Otherwise it will be a lot of noise in the log file.

Reply:
It might be useful to have a log line for when instance is not found without stack trace.